### PR TITLE
Document all remaining pre-trained weights

### LINKED ISSUE #6014: Improve documentation for Multi-weight support
### 📚 The doc issue

Previously we have restructured the documentation for models (#5833, #5987, #5897 etc), so that we can add necessary information about how the models were trained and integrate with the new Multi-weight support API.

This ticket is a tracker for all remaining tasks related to model documentation + Multi-weight support API:
- [x] Clean up the documentation pages #6003 #6002
- [x] Include information about the preprocessing transforms of models #6009
- [x] Add examples on how to use the new Multi-weight model builders #6013 
- [x] Move from the old `models.srt` any useful info to the new one (TORCH_HOME + TorchHub info, train/eval modes etc) #6026 #6025
- [x] Update galleries to use new idioms (for example fetching [classes](https://github.com/pytorch/vision/blob/11bd2eaa6d6976129836b329b01d1300babddcc9/gallery/plot_visualization_utils.py#L149) from meta-data and the text concerning [preprocessing](https://github.com/pytorch/vision/blob/11bd2eaa6d6976129836b329b01d1300babddcc9/gallery/plot_visualization_utils.py#L122-L124)) #6030
- [ ] Add description to pre-trained weights that clarify their scope #6036 #6039

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #6014

Similar to #6036, we document the weights for all remaining weights.

### CODE REVIEW

File: test/test_extended_models.py [hunk 1]
```diff
@@ -95,8 +95,8 @@ def test_schema_meta_validation(model_fn):
     # mandatory fields for each computer vision task
     classification_fields = {"categories", ("metrics", "acc@1"), ("metrics", "acc@5")}
     defaults = {
-        "all": {"metrics", "min_size", "num_params", "recipe"},
-        "models": classification_fields | {"_docs"},
+        "all": {"metrics", "min_size", "num_params", "recipe", "_docs"},
```
Comment (NicolasHug) [inline test/test_extended_models.py:98]: Now that this is mandatory for all weights, we can change these lines

https://github.com/pytorch/vision/blob/ae1d70713a542aac905cf403844a21b38b2be593/docs/source/conf.py#L348-L350

into 

```py
 lines += [meta_with_metrics.pop("_docs")]   # Custom per-Weights docs 
```

File: torchvision/models/detection/faster_rcnn.py [hunk 2]
```diff
@@ -454,7 +458,7 @@ def fasterrcnn_resnet50_fpn(
 ) -> FasterRCNN:
     """
     Faster R-CNN model with a ResNet-50-FPN backbone from the `Faster R-CNN: Towards Real-Time Object
-    Detection with Region Proposal Networks <https://arxiv.org/abs/1703.06870>`__
+    Detection with Region Proposal Networks <https://arxiv.org/abs/1506.01497>`__
```
Comment (NicolasHug) [inline torchvision/models/detection/faster_rcnn.py:461]: Nice catch!

**Review by NicolasHug — APPROVED** [review]
Thanks @datumbox , LGTM!

I haven't checked the rendering but I'm sure we'll catch major problems (if any) soon enough.

File: test/test_extended_models.py (hunk 1 above)
Comment (datumbox) [reply test/test_extended_models.py:98]: good spot done.

**Pull request merged**